### DISCUSSION TITLE
Systemd setup with pipx

### DISCUSSION THREAD

**Discussion by jonasprokopf** [Taxel/PlexTraktSync#1781] [Support] [answered]
Hey,
when I try to set up PlaxTraktWync via systemd, I run into this error: 

```root@bierkasten2:/etc/systemd/system# sudo systemctl status trakt.service
× trakt.service - PlexTraktSync watch daemon
     Loaded: loaded (/etc/systemd/system/trakt.service;>
     Active: failed (Result: exit-code) since Wed 2024->
   Main PID: 282262 (code=exited, status=203/EXEC)
        CPU: 4ms
```

It's looking like I need to change the ExecStart line, but since I've installed Plextraktsync via pipx, I don't think there is a traditional install directory with the .sh file? Any pointers would be much appreciated.

**Comment by KureOnGit** [comment]
On a Pi my install landed on /home/pi/.local/pipx/venvs/plextraktsync/bin/plextraktsync.
My systemmd service works since i added the full path to the ExecStart line

**Reply by jonasprokopf** [reply]
Thanks, for me the venvs directory is just empty unfortunately

**Reply by KureOnGit** [reply]
Try the "pipx list" @glensc mentioned in a thread I opened.
Should give you the directories you need

**Comment by glensc** [comment]
incomplete report, not reproducible. you not even showing the contents of your service file.

also, edit your post and reformat your report using markdown code fence:
- https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

**Reply by jonasprokopf** [reply]
Apologies, here's the .service file. The install directory was where @KureOnGit pointed out, I just looked in the wrong user directory last time:
```[Unit]
Description=PlexTraktSync watch daemon
After=network-online.target

[Service]
ExecStart=/home/szwab/.local/pipx/venvs/plextraktsync/bin/plextraktsync watch                                   Restart=on-failure
RestartSec=10
User=master
Group=master

[Install]
WantedBy=multi-user.target
```
Even with the full install path in the service pile, I get the same error as posted above.

**Reply by glensc** [reply, marked as answer]
"that above" is not useful.

use `journalctl -u <unit_name>` to see complete logs. alsom, when sharing command output, also share the command that was executed to produce that output.

does the command you put to execstart work from terminal?

also, I told KureOnGit multiple times not to use internal pipx path, but he does not listen. use path that pipx installs and creates wrapper.

- https://github.com/Taxel/PlexTraktSync/discussions/1769#discussioncomment-8329601

**Reply by jonasprokopf** [reply]
Okay, updating the execstart line with plextraktsync in the path directory worked. I also used the wrong user credentials, but now it's all set up. Thanks!

**Comment by glensc** [comment]
you might find systemd user mode instructions useful:
- https://github.com/Taxel/PlexTraktSync/pull/1782
